agent/exec/container/controller.go

```diff
@@ -362,7 +369,35 @@ func parseContainerStatus(ctnr types.ContainerJSON) (*api.ContainerStatus, error
 		ContainerID: ctnr.ID,
 		PID:         int32(ctnr.State.Pid),
 		ExitCode:    int32(ctnr.State.ExitCode),
+		Health:      parseHealthStatus(ctnr.State.Health),
 	}
 
 	return status, nil
 }
+
+func parseHealthStatus(health *types.Health) *api.HealthStatus {
+	if health == nil {
+		return nil
+	}
+	healthStatus := &api.HealthStatus{
+		Status:        health.Status,
+		FailingStreak: int32(health.FailingStreak),
+	}
+	// now parse the HealthcheckResult log
+	if health.Log != nil {
+		log := []*api.HealthcheckResult{}
+		for _, result := range health.Log {
+			startProto, _ := ptypes.TimestampProto(result.Start)
```

Comment:
Use `func MustTimestampProto(t time.Time) *tspb.Timestamp`. And you don't need this extra variable.